Add recipe for shanty-themes

qhga opened the pull request:
### Brief summary of what the package does

This package provides a dark and a light theme. 
The color scheme is based on dark blues and golden yellows.

### Direct link to the package repository

https://github.com/qhga/shanty-themes/tree/melpa

### Your association with the package

Maintainer

### Relevant communications with the upstream package maintainer

**None needed**

### Checklist

<!-- Please confirm with `x`: -->

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I have confirmed some of these without doing them

<!-- After submitting, please fix any problems the CI reports. -->

riscy commented:
Thanks for this!  Quick first pass --

`shanty-dark-theme.el` with package-lint 20220213.1457:
```
5 issues found:
1:0: error: There is no `provide' form.
10:0: error: Package-Requires outside the main file have no effect.
34:0: error: "shanty-colors" doesn't start with package's prefix "shanty-themes".
75:0: error: "shanty-dark" doesn't start with package's prefix "shanty-themes".
80:0: error: "shanty-dark" doesn't start with package's prefix "shanty-themes".
```
I think the `no provide` is a false positive since you call `provide-theme` but you can silence the false positive by adding an extra `(provide` block right after the `provide-theme` line.  The prefix lints are real considerations, though -- for this entire package your prefix should be shanty-themes, so maybe you want to go for "shanty-themes-dark" or "shanty-theme-dark".

`shanty-light-theme.el` with package-lint 20220213.1457:
```
5 issues found:
1:0: error: There is no `provide' form.
10:0: error: Package-Requires outside the main file have no effect.
34:0: error: "shanty-colors" doesn't start with package's prefix "shanty-themes".
75:0: error: "shanty-light" doesn't start with package's prefix "shanty-themes".
80:0: error: "shanty-light" doesn't start with package's prefix "shanty-themes".
```

Package and license:
- Avoid specifying `:branch` except in unusual cases; ideally you don't have a special `melpa` branch; MELPA would much prefer to pull from your mainline

qhga commented:
Hey @riscy,

thank you for your suggestions. For some reason - probably I am doing smth wrong - package-lint only showed similar errors in `shanty-themes.el` (which I fixed before I opened the pull request) but not in the other two files you mentioned. Anyways, I 

- removed the extra branch `melpa` and adjusted the recipe,
- added the additional `provide`, 
- removed the extra `Package-Requires` lines, 
- prefixed all variables and other definitons with `shanty-themes-` and 
- renamed the files so the theme is recognized properly. 
 
I hope this time I did not miss anything. Thanks for your help, it is highly appreciated!

riscy commented:
Thanks for accommodating!  One thing I may be doing differently is exercising the `package-lint-main-file` variable.  One way to make use of this variable is to add the following somewhere in each of your secondary files (just make sure that it isn't inside your `;;; Commentary` block):

```elisp
;; Local variables:
;; package-lint-main-file: "my-main-file.el"
;; end:
```

Looks good to merge now!
